### PR TITLE
fix NO_PROXY

### PR DESCRIPTION
**Pull request by BenTheElder**
- inspect the correct docker network
- include node names in no_proxy

### CODE REVIEW

File: pkg/cluster/internal/providers/docker/provision.go [hunk 1]
```diff
@@ -271,18 +283,20 @@ func runArgsForLoadBalancer(cfg *config.Cluster, name string, args []string) ([]
 	return append(args, loadbalancer.Image), nil
 }
 
-func getProxyEnv(cfg *config.Cluster) (map[string]string, error) {
+func getProxyEnv(cfg *config.Cluster, networkName string, nodeNames []string) (map[string]string, error) {
```
Comment (aojea) [inline pkg/cluster/internal/providers/docker/provision.go:286]: should `[]nodeNames` be in `common` and not provider specific?

Comment (BenTheElder) [reply pkg/cluster/internal/providers/docker/provision.go:286]: nah, there's no DNS in the others at the moment. we can DRY out later if that becomes the case

**Comment by BenTheElder** [comment]
hrm NFS flakes ... https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kind/1573/pull-kind-e2e-kubernetes/1258906611737432064

going to need to devote some time to test deflake very soon.

**Comment by mkimuram** [comment]
@BenTheElder 

Is my understanding right that in kind v0.8.x until this commit is included, or in kind v0.8.0 and v0.8.1, users need to specify `NO_PROXY` in a different way from what is described in [the current document](https://kind.sigs.k8s.io/docs/user/quick-start/#configure-kind-to-use-a-proxy), when deploying kind behind a proxy?

The document asks to add "172.17.0.0/16" to `NO_PROXY`, however these versions seem to require to add docker network's subnet and all nodes' name, like "172.18.0.0/16,kind-control-plane". (And after this commit is included, users won't need to specify anything special to `NO_PROXY` for changing kind's behavior.)

I've tested with v0.8.1, v0.8.0, v0.7.0, and master branch and reached to above conclusion, which took me a bit of time. If above is correct, I think that it is worth documented in somewhere. (If next version is planed to release soon, there won't be many people confused by this, though.)

BTW, this PR is very nice work, for it removed much complexity for users in deploying kind behind a  proxy. Thank you very much for your work!

**Comment by BenTheElder** [comment]
> The document asks to add "172.17.0.0/16" to NO_PROXY, however these versions seem to require to add docker network's subnet and all nodes' name, like "172.18.0.0/16,kind-control-plane". (And after this commit is included, users won't need to specify anything special to NO_PROXY for changing kind's behavior.)



That's correct, sorry about that.

The next release is tentatively due by june 1st, though we got a bit less done than I hoped so far .. https://github.com/kubernetes-sigs/kind/milestones

> BTW, this PR is very nice work, for it removed much complexity for users in deploying kind behind a proxy. Thank you very much for your work!

thanks :-)

**Pull request merged**